includes/storage/SQLStore/SMW_SQLStore3_SetupHandlers.php

```diff
@@ -305,123 +305,6 @@ public function drop( $verbose = true ) {
-			}
-		}
-
-		// update by internal SMW id --> make sure we get all objects in SMW
-		$dbr = wfGetDB( DB_SLAVE );
-
-		$res = $dbr->select(
-			SMWSql3SmwIds::tableName,
-			array( 'smw_id', 'smw_title', 'smw_namespace', 'smw_iw', 'smw_subobject' ),
-			array(
-				"smw_id >= $index ",
-				" smw_id < " . $dbr->addQuotes( $index + $count )
-			),
-			__METHOD__
-		);
-
-		foreach ( $res as $row ) {
-			$emptyrange = false; // note this even if no jobs were created
-
-			if ( $namespaces && !in_array( $row->smw_namespace, $namespaces ) ) {
-				continue;
-			}
-
-			// Find page to refresh, even for special properties:
-			if ( $row->smw_title != '' && $row->smw_title{0} != '_' ) {
-				$titleKey = $row->smw_title;
-			} elseif ( $row->smw_namespace == SMW_NS_PROPERTY && $row->smw_iw == '' && $row->smw_subobject == '' ) {
-				$titleKey = str_replace( ' ', '_', SMWDIProperty::findPropertyLabel( $row->smw_title ) );
-			} else {
-				$titleKey = '';
-			}
-
-			if ( $row->smw_subobject !== '' ) {
-				// leave subobjects alone; they ought to be changed with their pages
-			} elseif ( ( $row->smw_iw === '' || $row->smw_iw == SMW_SQL3_SMWREDIIW ) &&
-				$titleKey != '' ){
-				// objects representing pages
-				// TODO: special treament of redirects needed, since the store will
-				// not act on redirects that did not change according to its records
-				$title = Title::makeTitleSafe( $row->smw_namespace, $titleKey );
-
-				if ( $title !== null && !$title->exists() ) {
-					$updatejobs[] = new UpdateJob( $title );
-				}
-			} elseif ( $row->smw_iw == SMW_SQL3_SMWIW_OUTDATED ) { // remove outdated internal object references
-				$dbw = wfGetDB( DB_MASTER );
-				foreach ( $this->store->getPropertyTables() as $proptable ) {
-					if ( $proptable->usesIdSubject() ) {
-						$dbw->delete( $proptable->getName(), array( 's_id' => $row->smw_id ), __METHOD__ );
-					}
-				}
-
-				$dbw->delete( SMWSql3SmwIds::tableName, array( 'smw_id' => $row->smw_id ), __METHOD__ );
-			} elseif ( $titleKey != '' ) { // "normal" interwiki pages or outdated internal objects -- delete
-				$diWikiPage = new SMWDIWikiPage( $titleKey, $row->smw_namespace, $row->smw_iw );
-				$emptySemanticData = new SMWSemanticData( $diWikiPage );
-				$this->store->updateData( $emptySemanticData );
-			}
-		}
-		$dbr->freeResult( $res );
-
-		wfRunHooks('smwRefreshDataJobs', array(&$updatejobs));
-
-		if ( $usejobs ) {
-			JobBase::batchInsert( $updatejobs );
-		} else {
-			foreach ( $updatejobs as $job ) {
-				$job->run();
-			}
-		}
-
-		$nextpos = $index + $count;
-
-		if ( $emptyrange ) { // nothing found, check if there will be more pages later on
-			$next1 = $dbr->selectField( 'page', 'page_id', "page_id >= $nextpos", __METHOD__, array( 'ORDER BY' => "page_id ASC" ) );
-			$next2 = $dbr->selectField( SMWSql3SmwIds::tableName, 'smw_id', "smw_id >= $nextpos", __METHOD__, array( 'ORDER BY' => "smw_id ASC" ) );
-			$nextpos = $next2 != 0 && $next2 < $next1 ? $next2 : $next1;
```

Comment:
`$next2 < $next1` does not guarantee that all ids are fetched especially when pageIds (next1) are lower in count than the smwIds, and smwIds are expected to be higher in count (each single annotation is represented by an id) therefore `$next2 < $next1` will stop at the max pageId leaving anything beyond `$next2 >= $next1` untouched.